services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/directory_watcher.py

```diff
@@ -85,10 +85,17 @@ async def _push_directory_after_event_chain(directory_path: Path) -> None:
     await _push_directory(directory_path)
 
 
-def async_push_directory(event_loop: AbstractEventLoop, directory_path: Path) -> None:
-    event_loop.create_task(
-        _push_directory_after_event_chain(directory_path), name=TASK_NAME_FOR_CLEANUP
+def async_push_directory(
+    event_loop: AbstractEventLoop,
+    directory_path: Path,
+    tasks_collection: set[asyncio.Task[Any]],
+) -> None:
+    task = event_loop.create_task(
+        _push_directory_after_event_chain(directory_path),
+        name=TASK_NAME_FOR_CLEANUP,
     )
+    tasks_collection.add(task)
+    task.add_done_callback(tasks_collection.discard)
```

Comment:
I can already tell you that in this situation this will not work as you expect. 

You might get a lot of tasks spawned all at once, something like 100-1000 (this can happen when unzipping a directory with lots of files).

At that point the fist which finishes will remove some or all the tasks being tracked. I would go for a better way to remove them.

Maybe using a dict instead of a set and remove the task by name would be a better way to go when it is finished? And you would require to append something to the task names to make them unique at this point.

Reply:
I am not sure what you are talking about actually. a task is an object so why would everything be discarded. Also, this is the official recommended way from the python documentation. Can you put some code on what exactly should happen?
Also, if there are 1000 same tasks, why not accumulate them into just one instead of calling them all?

Reply:
see [this proof](https://github.com/sanderegg/osparc-simcore/blob/5b1095ed8b9a3e69ab5438ea8231efa28f3a55a2/packages/service-library/tests/test_utils.py)